charts/argo-rollouts/templates/argo-rollouts-clusterrole.yaml

```diff
@@ -29,6 +29,14 @@ rules:
   - list
   - watch
   - patch
+- apiGroups:
+    - ""
+  resources:
+    - secrets
```

Comment:
Is the intent here to give the `argo-rollouts-clusterrole` full read access to all cluster secrets?  If the intent of this PR is to add the ability to run the controller using images from a private registry, I think we can do without this?  I'm reading the docs, and it looks like we can add imagepull secrets to a service account if need be: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account

Reply:
This change comes from the AnalysisTemplates being able to reference secrets: https://github.com/argoproj/argo-rollouts/pull/420

This originally came up for use with the WebMetric provider, and I've recently been using it with the datadog metric provider I'm testing.